### PR TITLE
Adding completion for remaining Eventing resources

### LINKED ISSUE #1373: Context sensitive Bash completion
### Feature request

`kn completion` is currently based on the auto generation of the bash/zsh completion code by cobra. It lacks the the possibility to complete on context sensitive information likes service or revision names, something that e.g. kubectl allows. 

We should extend the completion code to allow completion on those resources. The way how `kubectl` does it could be taken as an inspiration.

### UI Example

```
kn service describe my<TAB>
```

would complete to "myservice" if there would be a Knative service installed that is called "myservice".

Here are the smaller issues tracked by this issue:

- [x] #1374 
- [x] #1547 
- [x] #1559
- [x] #1560
- [x] #1561
- [ ] #1562

### PR DESCRIPTION
**Pull request by vyasgun**
## Description
Context specific autocompletion for eventing resources

## Changes

* Added the ValidArgs function for the following resources
    * triggers
    * source:
        * ping
        * sinkbinding
        * apiserver
        * container
    * channel
    * subscription  

## Reference

Fixes #1373

### CODE REVIEW

File: pkg/kn/commands/completion_helper.go [hunk 1]
```diff
@@ -231,3 +240,206 @@ func completeDomain(config *completionConfig) (suggestions []string) {
...
+	if len(config.args) != 0 {
+		return
+	}
+	namespace, err := config.params.GetNamespace(config.command)
+	if err != nil {
+		return
+	}
+
+	var pingSourcesClient clientv1beta2.KnPingSourcesClient
+	if config.params.NewSourcesV1beta2Client == nil {
```
Comment (dsimansk) [inline pkg/kn/commands/completion_helper.go:359]: Is there any specific reason to initialize only `SourcesV1beta2Client`, but other completion functions don't follow this pattern?

Comment (dsimansk) [reply pkg/kn/commands/completion_helper.go:359]: Ehm, I see...it isn't properly done. Please fix that by adding it here:
https://github.com/knative/client/blob/main/pkg/kn/commands/types.go#L69

Comment (vyasgun) [reply pkg/kn/commands/completion_helper.go]: Thanks, fixed.

**Review by dsimansk — APPROVED** [review]
Thanks!

/approve
/lgtm

**Pull request merged**